src/main/java/org/phoebus/olog/ElasticConfig.java

```diff
@@ -199,17 +217,36 @@ public ElasticsearchClient getClient() {
                     ES_HTTP_CONNECT_TIMEOUT_MS,
                     ES_HTTP_SOCKET_TIMEOUT_MS
             ));
-            RestClient httpClient = RestClient.builder(new HttpHost(host, port, protocol))
+            RestClientBuilder clientBuilder = RestClient.builder(new HttpHost(host, port, protocol))
                     .setRequestConfigCallback(builder ->
                             builder.setConnectTimeout(ES_HTTP_CONNECT_TIMEOUT_MS)
                                     .setSocketTimeout(ES_HTTP_SOCKET_TIMEOUT_MS)
-                    )
-                    .setHttpClientConfigCallback(builder ->
-                            // Avoid timeout problems
-                            // https://github.com/elastic/elasticsearch/issues/65213
-                            builder.setKeepAliveStrategy((response, context) -> ES_HTTP_CLIENT_KEEP_ALIVE_TIMEOUT_MS)
-                    )
-                    .build();
+                    );
+
+            // Configure authentication
+            if (!authorizationHeader.isEmpty()) {
+                clientBuilder.setDefaultHeaders(new Header[] {new BasicHeader("Authorization", authorizationHeader)});
+                if (!username.isEmpty() || !password.isEmpty()) {
+                    logger.log(Level.WARNING, "elasticsearch.authorization.header is set, ignoring elasticsearch.authorization.username and elasticsearch.authorization.password.");
+                }
+            } else if (!username.isEmpty() || !password.isEmpty()) {
+                final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
+                clientBuilder.setHttpClientConfigCallback(httpClientBuilder ->
```

Comment:
This should be called outside the if block to make sure it is set regardless of authorization strategy.